common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java

```diff
@@ -163,35 +164,45 @@ public Mono<Void> runAllAsync(long endNanoTime) {
         lastCompletionNanoTime = 0;
         long startNanoTime = System.nanoTime();
 
-        return Flux.just(1)
-            .repeat()
-            .flatMap(i -> runTestAsync(), 1)
-            .doOnNext(v -> {
-                completedOperations += v;
+        return Flux.generate(sink -> {
+                if (System.nanoTime() < endNanoTime) {
+                    sink.next(1);
+                } else {
+                    sink.complete();
+                }
+            })
+            .flatMap(ignored -> {
```

Comment:
Why do we need to check if the timer expired twice?  The timer is approximate so running one extra operation should be fine.  Returning a result of "0" won't increase completedOperations, but it will update lastCompletionNanoTime, which seems wrong.

Reply:
I'll revert this in #32020